### PR TITLE
Create Test ProxyNetworkConfig

### PR DESCRIPTION
**Pull request by Mercurialf**

### CODE REVIEW

File: src/test/java/executor/service/model/ProxyNetworkConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public void testEmptyConstructor() {
+        ProxyNetworkConfig config = new ProxyNetworkConfig();
+        assertNull(config.getHostname());
+        assertNull(config.getPort());
+    }
+
+    @Test
+    public void testParameterizedConstructor() {
+        String hostname = "example.com";
+        Integer port = 8080;
```
Comment (jurok3x) [inline src/test/java/executor/service/model/ProxyNetworkConfigTest.java]: extract constants

File: src/test/java/executor/service/model/ProxyNetworkConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        Integer port = 8888;
+        config.setHostname(hostname);
+        config.setPort(port);
+
+        assertEquals(hostname, config.getHostname());
+        assertEquals(port, config.getPort());
+    }
+
+    @Test
+    public void testEqualsAndHashCode() {
```
Comment (jurok3x) [inline src/test/java/executor/service/model/ProxyNetworkConfigTest.java]: separate this test

File: src/test/java/executor/service/model/ProxyNetworkConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    public void testGettersAndSetters() {
+        ProxyNetworkConfig config = new ProxyNetworkConfig();
+        String hostname = "proxy.example.com";
+        Integer port = 8888;
+        config.setHostname(hostname);
+        config.setPort(port);
+
+        assertEquals(hostname, config.getHostname());
+        assertEquals(port, config.getPort());
```
Comment (jurok3x) [inline src/test/java/executor/service/model/ProxyNetworkConfigTest.java]: no need to test separate fields when object equal() doing it all.
see lecture video.

**Review by jurok3x — CHANGES_REQUESTED** [review]

File: src/test/java/executor/service/model/ProxyNetworkConfigTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    private static final Integer PORT = 8080;
+    private static final ProxyNetworkConfig CONFIG1;
+    private static final ProxyNetworkConfig CONFIG2;
+    private static final ProxyNetworkConfig CONFIG3;
+
+    static {
+        CONFIG1 = new ProxyNetworkConfig("example.com", 8080);
+        CONFIG2 = new ProxyNetworkConfig("example.com", 8080);
+        CONFIG3 = new ProxyNetworkConfig("example.org", 8081);
+    }
```
Comment (hromovyi-kostiantyn) [inline src/test/java/executor/service/model/ProxyNetworkConfigTest.java]: Better do beforeEach method

**Comment by OlexTul** [comment]
package executor.service.model;

import org.junit.Test;

import static org.junit.Assert.*;

public class ProxyNetworkConfigTest {

    @Test
    public void testEqualsAndHashCode() {
        ProxyNetworkConfig config1 = new ProxyNetworkConfig("example.com", 8080);
        ProxyNetworkConfig config2 = new ProxyNetworkConfig("example.com", 8080);
        ProxyNetworkConfig config3 = new ProxyNetworkConfig("different.com", 8080);

        // Test equals method
        assertEquals(config1, config2);
        assertNotEquals(config1, config3);

        // Test hashCode method
        assertEquals(config1.hashCode(), config2.hashCode());
        assertNotEquals(config1.hashCode(), config3.hashCode());
    }

    @Test
    public void testGettersAndSetters() {
        ProxyNetworkConfig config = new ProxyNetworkConfig();

        config.setHostname("example.com");
        config.setPort(8080);

        assertEquals("example.com", config.getHostname());
        assertEquals(Long.valueOf(8080L), Long.valueOf(config.getPort()));
    }
}

**Comment by OlexTul** [comment]
I would do like as :
package executor.service.model;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;
import static org.junit.Assert.assertEquals;

public class ProxyNetworkConfigTest {
    private static final String HOSTNAME = "example.com";
    private static final String HOSTNAME_ANOTHER = "example.org";
    private static final Integer PORT = 8080;
    private static final Integer PORT_ANOTHER = 8081;
    private ProxyNetworkConfig config1;
    private ProxyNetworkConfig config2;
    private ProxyNetworkConfig config3;

    @Before
    public void setUp() {
        config1 = new ProxyNetworkConfig(HOSTNAME, PORT);
        config2 = new ProxyNetworkConfig(HOSTNAME, PORT);
        config3 = new ProxyNetworkConfig(HOSTNAME_ANOTHER, PORT_ANOTHER);
    }

    @After
    public void tearDown() {
        config1 = null;
        config2 = null;
        config3 = null;
    }

    @Test
    public void testEmptyConstructor() {
        ProxyNetworkConfig config = new ProxyNetworkConfig();
        assertNull(config.getHostname());
        assertNull(config.getPort());
    }

    @Test
    public void testParameterizedConstructorAndSetters() {
        ProxyNetworkConfig config = new ProxyNetworkConfig();
        config.setHostname(HOSTNAME);
        config.setPort(PORT);

        assertEquals(config, config1);
    }

    @Test
    public void testEquals() {
        assertEquals(config1, config2);
        assertNotEquals(config1, config3);
    }

    @Test
    public void testHashCode() {
        assertEquals(config1.hashCode(), config2.hashCode());
        assertNotEquals(config1.hashCode(), config3.hashCode());
    }
}

**Pull request merged**